src/wallet/wallet.cpp

```diff
@@ -7756,72 +7815,49 @@ std::optional<TransactionStrategy> TransactionStrategy::FromString(std::string p
+            return "AllowRevealedSenders";
         case PrivacyPolicy::AllowFullyTransparent:
+            return "AllowFullyTransparent";
         case PrivacyPolicy::AllowLinkingAccountAddresses:
+            return "AllowLinkingAccountAddresses";
         case PrivacyPolicy::NoPrivacy:
-            return true;
+            return "NoPrivacy";
         default:
-            // Fail closed.
-            return false;
+            assert(false);
     }
 }
 
-bool TransactionStrategy::AllowRevealedRecipients() {
-    switch (privacy) {
-        case PrivacyPolicy::FullPrivacy:
-        case PrivacyPolicy::AllowRevealedAmounts:
-        case PrivacyPolicy::AllowRevealedSenders:
-        case PrivacyPolicy::AllowLinkingAccountAddresses:
-            return false;
-        case PrivacyPolicy::AllowRevealedRecipients:
-        case PrivacyPolicy::AllowFullyTransparent:
-        case PrivacyPolicy::NoPrivacy:
-            return true;
-        default:
-            // Fail closed.
-            return false;
-    }
+bool TransactionStrategy::AllowRevealedAmounts() const {
+    return IsCompatibleWith(PrivacyPolicy::AllowRevealedAmounts);
 }
 
-bool TransactionStrategy::AllowRevealedSenders() {
-    switch (privacy) {
-        case PrivacyPolicy::FullPrivacy:
-        case PrivacyPolicy::AllowRevealedAmounts:
-        case PrivacyPolicy::AllowRevealedRecipients:
-            return false;
-        case PrivacyPolicy::AllowRevealedSenders:
-        case PrivacyPolicy::AllowFullyTransparent:
-        case PrivacyPolicy::AllowLinkingAccountAddresses:
-        case PrivacyPolicy::NoPrivacy:
-            return true;
-        default:
-            // Fail closed.
-            return false;
-    }
+bool TransactionStrategy::AllowRevealedRecipients() const {
+    return IsCompatibleWith(PrivacyPolicy::AllowRevealedRecipients);
 }
 
-bool TransactionStrategy::AllowLinkingAccountAddresses() {
-    switch (privacy) {
-        case PrivacyPolicy::FullPrivacy:
-        case PrivacyPolicy::AllowRevealedAmounts:
-        case PrivacyPolicy::AllowRevealedRecipients:
-        case PrivacyPolicy::AllowRevealedSenders:
-        case PrivacyPolicy::AllowFullyTransparent:
-            return false;
-        case PrivacyPolicy::AllowLinkingAccountAddresses:
-        case PrivacyPolicy::NoPrivacy:
-            return true;
-        default:
-            // Fail closed.
-            return false;
-    }
+bool TransactionStrategy::AllowRevealedSenders() const {
+    return IsCompatibleWith(PrivacyPolicy::AllowRevealedSenders);
+}
+
+bool TransactionStrategy::AllowFullyTransparent() const {
+    return IsCompatibleWith(PrivacyPolicy::AllowFullyTransparent);
+}
+
+bool TransactionStrategy::AllowLinkingAccountAddresses() const {
+    return IsCompatibleWith(PrivacyPolicy::AllowLinkingAccountAddresses);
+}
+
+bool TransactionStrategy::IsCompatibleWith(PrivacyPolicy requiredLevel) const {
+    return requestedLevel == PrivacyPolicyMeet(requestedLevel, requiredLevel);
```

Comment:
This is correct because
```
requestedLevel = requestedLevel ∧ requiredLevel <=> requestedLevel ≤ requiredLevel
```
[by definition](https://en.wikipedia.org/wiki/Join_and_meet#Universal_algebra_approach), and the RHS means "requestedLevel allows at least everything that requiredLevel allows", which as as intended.

Reply:
I could rename `requiredLevel` to just `policy` or `privacyPolicy`? I think the name just matched the way it’s used at the call site (in a later PR), and doesn’t apply to this general definitions.